### PR TITLE
Move some documentation from README to docusaurus

### PR DESCRIPTION
**Pull request by mihaibudiu**
I have also simplified the structure of the documentation pages, removing some redundant links.

### CODE REVIEW

**Comment by mihaibudiu** [comment]
I have tried to write a simple Rust program to invoke the generated SQL code, but I failed.
Help there would be appreciated.

**Comment by mihaibudiu** [comment]
Fixes #156 by checking in a link to the presentation.

File: docs/docs/guides/overview.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Using DBSP
+
+DBSP can be used in two ways:
+
+* You can use the Rust DBSP implementation and write directly Rust
+  programs that perform stream processing
+
+* You can write SQL programs and use our SQL compiler to generate
```
Comment (gz) [inline docs/docs/guides/overview.md:8]: we should add guides aimed at end-users. writing rust dbsp programs or invoking the compiler directly is for advanced users or dbsp developers and hopefully not needed by normal users. Instead I think the guides should teach these two things:

- How to create and run a dbsp pipeline with the python API
- How to create and run a dbsp pipeline with the UI

we can keep your guides too but if they won't be used by 90% of the users maybe it's better to put them under a separate category e.g., under Contribute to DBSP where we can teach those things?

File: docs/docs/sql/structure.md [hunk 2]
```diff
@@ -1,11 +0,0 @@
-# High-level program structure
```
Comment (gz) [inline docs/docs/sql/structure.md:1]: Why was this removed?

File: docs/docusaurus.config.js [hunk 3]
```diff
@@ -71,30 +71,12 @@ const config = {
...
           src: "img/logo.svg",
         },
         items: [
-          {
-            type: "docSidebar",
-            sidebarId: "docsSidebar",
-            position: "right",
-            label: "Get Started",
-          },
-          {
```
Comment (gz) [inline docs/docusaurus.config.js:87]: I think it's good to keep a Link to the REST API as a shortcut in the top navigation. Many users will write programs against that API and will need to access it to look something up.

**Review by gz — APPROVED** [review]

File: docs/docs/guides/overview.md (hunk 1 above)
Comment (mihaibudiu) [reply docs/docs/guides/overview.md:8]: At this point I don't know how to write a guide for users, so I write a guide for developers.
All are useful.

File: docs/docusaurus.config.js (hunk 3 above)
Comment (mihaibudiu) [reply docs/docusaurus.config.js:87]: I moved it to the left sidebar. We can reconsider later.

File: docs/docs/sql/structure.md (hunk 2 above)
Comment (mihaibudiu) [reply docs/docs/sql/structure.md:1]: This information is duplicated in the guide.
